temporalio/failure.py

```diff
@@ -0,0 +1,61 @@
+from enum import IntEnum
+from typing import Any, List, Optional
+
+import temporalio.api.enums.v1
+import temporalio.api.failure.v1
+import temporalio.converter
+
+
+class FailureError(Exception):
+    @staticmethod
+    async def from_proto(
+        failure: temporalio.api.failure.v1.Failure,
+        data_converter: temporalio.converter.DataConverter,
+    ) -> "FailureError":
+        raise NotImplementedError
+
+    def __init__(self, message: str, failure: "Failure") -> None:
+        super().__init__(message)
+        raise NotImplementedError
+
+
+class Failure:
+    def __init__(
+        self,
+        *details: Any,
+        proto_failure: Optional[temporalio.api.failure.v1.Failure] = None
+    ) -> None:
+        raise NotImplementedError
+
+
+class CancelledFailure(Failure):
+    def __init__(self, *details: Any) -> None:
+        super().__init__(*details)
+        raise NotImplementedError
+
+
+class TerminatedFailure(Failure):
+    def __init__(self, *details: Any, reason: Optional[str]) -> None:
+        super().__init__(*details)
+        raise NotImplementedError
+
+
+class TimeoutType(IntEnum):
```

Comment:
I'd probably put this in `common`

Reply:
Can you give more reasons why on these suggestions? Is it just because we did it in TypeScript?

I don't think it'll be used anywhere other than for this very specific error. It technically is referenced in proto history, but I don't expect us to be translating that to Python enums.

Where else is it used in TypeScript that makes it deserve to be in a "common" area? Why was it put into TypeScript `common` package to begin with?